src/main/java/com/softserve/academy/spaced/repetition/controller/CourseCommentController.java

```diff
@@ -34,70 +35,60 @@
         List<Comment> courseCommentsList = courseCommentService.getAllCommentsForCourse(courseId);
         Link collectionLink = linkTo(methodOn(CourseCommentController.class)
-                .getAllCommentsByCourse(categoryId, courseId)).withSelfRel();
+                .getAllCommentsByCourse(courseId)).withSelfRel();
         List<CommentDTO> listOfComments = DTOBuilder
                 .buildDtoListForCollection(courseCommentsList,CommentDTO.class, collectionLink);
         List<CommentDTO> commentsTree = CommentDTO.buildCommentsTree(listOfComments);
         return new ResponseEntity<>(commentsTree, HttpStatus.OK);
     }
 
     @Auditable(action = AuditingAction.VIEW_COMMENT_FOR_COURSE)
-    @GetMapping(value = "/api/category/{categoryId}/courses/{courseId}/comments/{courseCommentId}")
-    public ResponseEntity<CommentDTO> getCommentByCourse(@PathVariable Long categoryId,
-                                                         @PathVariable Long courseId,
+    @GetMapping(value = "/{courseCommentId}")
+    public ResponseEntity<CommentDTO> getCommentByCourse(@PathVariable Long courseId,
                                                          @PathVariable Long courseCommentId) {
         LOGGER.debug("View comment with id {} for course with id: {}", courseCommentId, courseId);
         CourseComment courseComment = courseCommentService.getCommentById(courseCommentId);
         Link selfLink = linkTo(methodOn(CourseCommentController.class)
-                .getCommentByCourse(categoryId, courseId, courseCommentId)).withSelfRel();
+                .getCommentByCourse(courseId, courseCommentId)).withSelfRel();
         CommentDTO courseCommentDTO = DTOBuilder.buildDtoForEntity(courseComment, CommentDTO.class, selfLink);
         return new ResponseEntity<>(courseCommentDTO, HttpStatus.OK);
     }
 
     @Auditable(action = AuditingAction.CREATE_COMMENT_FOR_COURSE)
-    @PostMapping(value = "/api/category/{categoryId}/courses/{courseId}/comments")
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToCourse(#categoryId, #courseId)")
-    public ResponseEntity<CommentDTO> addCommentByCourse(@Validated(Request.class) @RequestBody
-                                                                 ReplyToCommentDTO replyToCommentDTO,
-                                                         @PathVariable Long categoryId,
+    @PostMapping(value = "/comments")
+    public ResponseEntity<CommentDTO> addCommentByCourse(@Validated(Request.class) @RequestBody ReplyToCommentDTO replyToCommentDTO,
                                                          @PathVariable Long courseId) throws NotAuthorisedUserException {
         LOGGER.debug("Added comment to course with id: {}", courseId);
         CourseComment courseComment = courseCommentService
                 .addCommentForCourse(courseId, replyToCommentDTO.getCommentText(), replyToCommentDTO.getParentCommentId());
```

Comment:
Maybe better to change input parameters of method `addCommentForCourse`. It seems that this method should take `replyToCommentDTO` as whole object, not parameters of it.